controllers/fetchers/jsonnet_fetcher.go

```diff
@@ -0,0 +1,40 @@
+package fetchers
+
+import (
+	"embed"
+	"errors"
+	"fmt"
+	"github.com/google/go-jsonnet"
+	"github.com/grafana-operator/grafana-operator-experimental/api/v1beta1"
+	"io/fs"
+	"os"
+	"strings"
+)
+
+func FetchJsonnet(dashboard *v1beta1.GrafanaDashboard, libsonnet embed.FS, libsonnetPath string) ([]byte, error) {
+	vm := jsonnet.MakeVM()
+
+	if dashboard.Spec.Jsonnet == "" {
+		return nil, errors.New("no jsonnet Content Found, nil or empty string")
+	}
+
+	_, err := os.Stat(libsonnetPath)
```

Comment:
Like you mentioned, this will need to be adjusted as `os` package looks into "real" filesystem, not into the virtualized one.
As I see, `embed` implements Stat() ([link](https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/embed/embed.go;drc=15605ca827723d568f780402d03b29842fd20eec;l=353)), so I guess you need to replace it with fs.Stat or something like that :)